File: Safari-611.1.10/LayoutTests/http/tests/misc/resources/char-encoding-in-hidden-charset-field.php
<?php
header("Content-type: text/html; charset=UTF-8");
?>
<html>
<head>
<title>Test for bug 19079</title>
</head>
<body>
<p>
This is a test for https://bugs.webkit.org/show_bug.cgi?id=19079, it send the submissions
character encoding in hidden _charset_ field.
</p>

<?php
if ($_SERVER['REQUEST_METHOD'] == 'POST') {
$data = file_get_contents("php://input");
if ($data != "")
echo "<p>PASSED: $data </p>";
else
echo "<p>PASSED: No _charset_ field </p>";
}
if ($_SERVER['REQUEST_METHOD'] == 'GET') {
$data = htmlentities($_GET['_charset_']);
echo "<p>PASSED: $data </p>";
}
?>

<script>
if(window.testRunner)
    testRunner.notifyDone();
</script>

</body>
</html>

File: LayoutTests/http/tests/misc/resources/char-encoding-in-hidden-charset-field.py
#!/usr/bin/env python3

import os
import sys
from urllib.parse import parse_qs

request_method = os.environ.get('REQUEST_METHOD', '')
charset = parse_qs(os.environ.get('QUERY_STRING', ''), keep_blank_values=True).get('_charset_', [''])[0]

data = ''
if request_method == 'POST':
    data = ''.join(sys.stdin.readlines())

sys.stdout.write(
    'Content-Type: text/html; charset=UTF-8\r\n\r\n'
    '<html>\n'
    '<head>\n'
    '<title>Test for bug 19079</title>\n'
    '</head>\n'
    '<body>\n'
    '<p>\n'
    'This is a test for https://bugs.webkit.org/show_bug.cgi?id=19079, it send the submissions\n'
    'character encoding in hidden _charset_ field.\n'
    '</p>\n'
)

if request_method == 'POST':
    if data:
        sys.stdout.write('<p>PASSED: {} </p>'.format(data))
    else:
        sys.stdout.write('<p>PASSED: No _charset_ field </p>')

if request_method == 'GET':
    sys.stdout.write('<p>PASSED: {} </p>'.format(charset))

sys.stdout.write(
    '<script>\n'
    'if(window.testRunner)\n'
    '    testRunner.notifyDone();\n'
    '</script>\n'
    '\n'
    '</body>\n'
    '</html>\n'
)